tests/standalone_tests/test_guard.py

```diff
@@ -0,0 +1,219 @@
+
+
+def test_is_lowercase_identifier_valid():
+    assert is_lowercase_identifier("some_name_1")
+
+
+@pytest.mark.parametrize(
+    "string",
+    [
+        "SomeName",
+        "1some_name",
+        "some.name",
+    ],
+)
+def test_is_lowercase_identifier_invalid(string):
+    assert not is_lowercase_identifier(string)
+
+
+@pytest.mark.parametrize(
+    "string",
+    [
+        "0.0.0.0:1",
+        "192.168.1.1:50000",
+        "255.255.255.255:65535",
+    ],
+)
+def test_is_socket_address_valid(string):
+    assert is_socket_address(string)
+
+
+@pytest.mark.parametrize(
+    "string",
+    [
+        "0.0.0.0:0",
+        "192.168.1:50000",
+        " 192.168.1.1:50000",
+        "192.168.1.1:50000 ",
+        "192.168.1.1",
+        "255.255.255.256:65535",
+        "255.255.255.255:65536",
+    ],
+)
+def test_is_socket_address_invalid(string):
+    assert not is_socket_address(string)
+
+
+@pytest.mark.parametrize(
+    "string",
+    [
+        "datamodel:0",
+        "datamodel:0.1",
+        "datamodel:alpha",
+        "datamodel:alpha_beta",
+        "data_model_1:0.1",
+    ],
+)
+def test_is_datamodel_valid(string):
+    assert is_datamodel(string)
+
+
+@pytest.mark.parametrize(
+    "string",
+    [
+        "datamodel",
+        "datamodel:",
+        ":0.1",
+        "datamodel:.0",
+        "datamodel:0.",
+    ],
+)
+def test_is_datamodel_invalid(string):
+    assert not is_datamodel(string)
+
+
+@pytest.mark.parametrize(
+    "string",
+    [
+        "data_table",
+        '"datamodel:0.1".data_table',
+        '"datamodel:0.1"."Data_Table"',
+    ],
+)
+def test_is_primary_data_table_valid(string):
+    assert is_primary_data_table(string)
+
+
+@pytest.mark.parametrize(
+    "string",
+    [
+        "data table",
+        '"datamodel:0.1".Data_Table',
+        '"datamodel:0.1".',
+        '"datamodel:0.1"."Data Table"',
+    ],
+)
+def test_is_primary_data_table_invalid(string):
+    assert not is_primary_data_table(string)
+
+
+def test_is_list_of_identifiers():
+    assert is_list_of_identifiers(["name_1", "name_2"])
+    assert not is_list_of_identifiers(["Name_1", "name_2"])
+
+
+def test_is_valid_filter():
+    assert is_valid_filter({"rules": [{"id": "name"}, {"rules": [{"id": "name"}]}]})
+    assert not is_valid_filter({"rules": [{"id": "name"}, {"rules": [{"id": "Name"}]}]})
+
+
+def test_is_valid_table_schema():
+    Column = namedtuple("Column", "name")
+    Schema = namedtuple("Schema", "columns")
+
+    valid_schema = Schema(columns=[Column(name="name_1"), Column(name="name_2")])
+    invalid_schema = Schema(columns=[Column(name="name_1"), Column(name="Name_2")])
+
+    assert is_valid_table_schema(valid_schema)
+    assert not is_valid_table_schema(invalid_schema)
+
+
+def test_sql_injection_guard__validate_posarg():
+    @sql_injection_guard(a=str.isalpha)
+    def f(a):
+        pass
+
+    try:
+        f("a")
+    except InvalidSQLParameter as exc:
```

Comment:
I would suggest catching any exception here.

Reply:
Why? I am interested in the behavior with respect to this particular exception. Of course, if any other exception is raised there is a bug somewhere and it has to be fixed, it's just not the bug I am testing for here. Catching this particular exception tells the reader that this test verifies that indeed, the parameter is **not** an invalid SQL parameter.

Reply:
What you are testing here is that this is a valid posarg for the function. So when you call it no exceptions should be raised hence the message: `pytest.fail(f"Unexpected exception {exc}")`

Any exception raised is an unexpected exception. Why would you need to change this test if the logic changes a bit in the validator?

Anyway, not important. Up to you.